### PR TITLE
Combined Bookmarks / "Saved posts" and history page

### PR DESCRIPTION
**Pull request by b0b3rt**
_Co-Authored-By: Sarah Cheng [sarah.cheng@centreforeffectivealtruism.org ](mailto:sarah.cheng@centreforeffectivealtruism.org)_

This PR turns the [Bookmarks](https://www.lesswrong.com/bookmarks) / [Saved posts](https://forum.effectivealtruism.org/saved) page into a page called "Saved & Read". Bookmarks is the primary tab, we moved the "Read history" page into the second tab, and the third tab is your vote history (posts and comments).

This is a continuation of a hackathon project ([figma here](https://www.figma.com/file/mqjkXkxOVVLOlyc0D3NDEm/%F0%9F%A6%A6-Hackathon?node-id=25%3A1369&mode=dev)) and https://github.com/ForumMagnum/ForumMagnum/pull/7029, so it's not perfect but feels good enough now to deploy and build off of later.

On the EA Forum:
"Saved" is the same as the "Saved posts" page
Screen Shot 2023-07-13 at 1 07 48 PM

"Read history" is the same as the "Read history" page
Screen Shot 2023-07-13 at 1 08 00 PM

"Vote history" is a mix of comments and post items, and we made a variant of our post item that includes an interactive vote component
Screen Shot 2023-07-13 at 1 08 08 PM

-----
On LW:
"Bookmarks" is the same as the "Bookmarks" page
Screen Shot 2023-07-13 at 12 43 49 PM

"Read history" is similar to EAF's
Screen Shot 2023-07-14 at 12 51 11 PM

"Vote history" is also similar to EAF's, except we didn't make a version of the LW post item with interactive voting
Screen Shot 2023-07-13 at 12 44 08 PM

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204838540402830) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by s-cheng** [comment]
@darkruby501 I built a bunch of this with @b0b3rt with the goal to make it work for both our sites. I think it's fine to ship this to LW as-is, but a couple things you might want to do here are:
1. Decide what the page should be called
  a. I left it as "Bookmarks" for everything (ex. page title, links) except for the heading at the top of the page
  b. Feel free to change the heading text from "Saved & Read" to something else
2. Update the LW post item to have interactive voting arrows in the "Vote history" tab

File: packages/lesswrong/components/bookmarks/BookmarksList.tsx [hunk 1]
```diff
@@ -35,15 +37,17 @@ const BookmarksList = ({limit=20, hideLoadMore=false}: {
...
-  return <div>
-    {sortedBookmarkedPosts && sortedBookmarkedPosts.map((post: PostsListWithVotes, i: number) =>
-      <PostsItem
-        key={post._id} post={post} bookmark
-        showBottomBorder={i < sortedBookmarkedPosts.length-1}
-      />
-    )}
-    {!hideLoadMore && <LoadMore {...loadMoreProps}/>}
-  </div>
+  return <AnalyticsContext pageSubSectionContext="bookmarksList">
```
Comment (s-cheng) [inline packages/lesswrong/components/bookmarks/BookmarksList.tsx:40]: Updated some analytics stuff to make it consistent across tabs

File: packages/lesswrong/components/bookmarks/ReadHistoryTab.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import React, {useState} from 'react';
```
Comment (s-cheng) [inline packages/lesswrong/components/bookmarks/ReadHistoryTab.tsx:1]: This file should mostly be the same as `ReadHistoryPage.tsx`

File: packages/lesswrong/components/ea-forum/users/EAUsersProfileLinks.tsx [hunk 3]
```diff
@@ -84,11 +84,6 @@ const EAUsersProfileLinks = ({user, classes}: {
           Manage subscriptions
         </Link>
       }
-      {currentUser && currentUser._id === user._id &&
-        <Link to="/history">
-          Read history
-        </Link>
-      }
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/users/EAUsersProfileLinks.tsx:91]: Removed this link to clean up the EAF user profile a bit

File: packages/lesswrong/components/users/UsersMenu.tsx [hunk 4]
```diff
@@ -263,6 +262,7 @@ const UsersMenu = ({classes}: {
                 iconClassName={classes.icon}
               />
             }
+            {isEAForum && messagesNode}
```
Comment (s-cheng) [inline packages/lesswrong/components/users/UsersMenu.tsx:265]: Slightly reordered the user menu options on EAF, so that "Saved & read" is right under "User profile" (I think that those two conceptually go together)

**Review by s-cheng — COMMENTED** [review]
There's also a few things we might want to change about the "Vote history" tab in the future:
1. Maybe exclude the user's votes on their own posts/comments
2. See if we can exclude deleted/draft content (might be annoying)
3. Think about how we want to handle extended vote types - like, should we include emoji reacts here? Not sure

File: packages/lesswrong/lib/routes.ts [hunk 5]
```diff
@@ -775,16 +775,10 @@ const forumSpecificRoutes = forumSelect<Route[]>({
...
     {
-      name: 'savedPosts',
+      name: 'savedAndRead',
       path: '/saved',
       componentName: 'BookmarksPage',
-      title: 'Saved Posts',
-    },
-    {
-      name: 'readHistory',
-      path: '/history',
```
Comment (oetherington) [inline packages/lesswrong/lib/routes.ts:785]: Maybe we should add a redirect from /history to /saved? (Might not be important though - I never realised /history existed, but maybe somebody uses it)

File: packages/lesswrong/components/posts/EAPostsItem.tsx [hunk 6]
```diff
@@ -322,12 +344,12 @@ const EAPostsItem = ({classes, ...props}: EAPostsItemProps) => {
             </div>
             <InteractionWrapper className={classes.interactionWrapper}>
               <PostsItemTrailingButtons
+                showArchiveButton={false}
```
Comment (oetherington) [inline packages/lesswrong/components/posts/EAPostsItem.tsx:347]: What's the reason for this change? (Not necessarily saying it's wrong)

File: packages/lesswrong/components/bookmarks/BookmarksPage.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  const [activeTab, setActiveTab] = useState<TabType>('bookmarks')
+  
+  useEffect(() => {
+    // unfortunately the hash is unavailable on the server, so we check it here instead
+    if (location.hash === '#readhistory') {
+      setActiveTab('readhistory')
+    } else if (location.hash === '#votehistory') {
+      setActiveTab('votehistory')
+    }
+    //No exhaustive deps because this is supposed to run only on mount
```
Comment (oetherington) [inline packages/lesswrong/components/bookmarks/BookmarksPage.tsx:52]: I'm not convinced by this - if I'm already on the page and I manually change the hash from "readhistory" to "votehistory", I'd expect the tab to change

File: packages/lesswrong/components/bookmarks/ReadHistoryTab.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+          ...PostsListWithVotes
+          lastVisitedAt
+        }
+      }
+    }
+    ${fragmentTextForQuery("PostsListWithVotes")}
+    `,
+    {
+      ssr: true,
+      fetchPolicy: "cache-and-network",
```
Comment (oetherington) [inline packages/lesswrong/components/bookmarks/ReadHistoryTab.tsx:40]: Not super important but I noticed that this query never seems to be cached - if you navigate between the tabs then the other two will be display imeditely the second time you visit them, but this one always loads.

Probably not worth spending a ton of time on. I'd guess it probably predates this PR too.

**Review by oetherington — APPROVED** [review]
LGTM

File: packages/lesswrong/components/posts/EAPostsItem.tsx (hunk 6 above)
Comment (s-cheng) [reply packages/lesswrong/components/posts/EAPostsItem.tsx:347]: This is `true` by default, which meant that there was an archive icon that appeared to the right of the post item on hover if the post was a draft. Clicking on the icon archives the draft post. You can see this if you view a user's "My Drafts" list in their profile. We don't have good support for letting users organize their drafts in general (ex. we don't have the sorting/filtering options that LW does), and I would guess that archiving your draft on the EA Forum is basically a black hole, so I figured we should just hide that icon for now.

File: packages/lesswrong/lib/routes.ts (hunk 5 above)
Comment (s-cheng) [reply packages/lesswrong/lib/routes.ts:785]: Yeah it was pretty hidden so I don't think it's worth adding
